### PR TITLE
Core: Fix webpack stats

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: #14139

## What I did

Forgot to push this final commit on other PR (!)

### CODE REVIEW

**Review by shilman — CHANGES_REQUESTED** [review]
CI is failing due to something stats related. can you take a look @tmeasday ?

**Review by shilman — APPROVED** [review]
Aha, LGTM 🙏

File: examples/official-storybook/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "do-storyshots-puppeteer": "../../node_modules/.bin/jest --projects=./storyshots-puppeteer",
     "generate-addon-jest-testresults": "jest --config=tests/addon-jest.config.json --json --outputFile=stories/addon-jest.testresults.json",
     "graphql": "node ./graphql-server/index.js",
-    "packtracker": "yarn storybook --smoke-test --webpack-stats-json --quiet && cross-env PT_PROJECT_TOKEN=1af1d41b-d737-41d4-ac00-53c8f3913b53 packtracker-upload --stats=./node_modules/.cache/storybook/public/manager-stats.json",
+    "packtracker": "yarn storybook --smoke-test --webpack-stats-json /tmp --quiet && cross-env PT_PROJECT_TOKEN=1af1d41b-d737-41d4-ac00-53c8f3913b53 packtracker-upload --stats=/tmp/manager-stats.json",
```
Comment (ndelangen) [inline examples/official-storybook/package.json:11]:
```suggestion
"packtracker": "yarn storybook --smoke-test --webpack-stats-json=/tmp --quiet && cross-env PT_PROJECT_TOKEN=1af1d41b-d737-41d4-ac00-53c8f3913b53 packtracker-upload --stats=/tmp/manager-stats.json",
```

File: lib/builder-webpack4/src/index.ts [hunk 2]
```diff
@@ -39,6 +39,13 @@ export const executor = {
   get: webpack,
 };
 
+export const makeStatsFromError: (err: string) => Stats = (err) =>
+  ({
+    hasErrors: () => true,
+    hasWarnings: () => false,
+    toJson: () => ({ warnings: [] as any[], errors: [err] }),
+  } as any);
```
Comment (ndelangen) [inline lib/builder-webpack4/src/index.ts:47]: Is this something we want to export?

In the future we'll want to map the builders internal stats object to something common for all builders.
Right now we're leaking webpack details from our 2 builders. This is a known issue, but we should be really careful to not make that situation worse.

Comment (ndelangen) [reply lib/builder-webpack4/src/index.ts:47]: I'm thinking something like SyntacticEvent in React, where the underlaying raw event is still available, but there's a layer on top to smooth out the browser differences. (except we're smoothing out the builder differences)

**Review by ndelangen — APPROVED** [review]

File: lib/builder-webpack4/src/index.ts (hunk 2 above)
Comment (tmeasday) [reply lib/builder-webpack4/src/index.ts:47]: Yes, I think that's the plan for 6.3 (or whenever we add our next builder).

We probably shouldn't be exporting this function however, that was an oversight.

File: lib/core-server/src/cli/dev.ts [hunk 3]
```diff
@@ -42,11 +42,7 @@ export async function getDevCli(packageJson: {
     .option('--docs-dll', 'Use Docs dll reference (legacy)')
     .option('--ui-dll', 'Use UI dll reference (legacy)')
     .option('--debug-webpack', 'Display final webpack configurations for debugging purposes')
-    .option(
-      '--webpack-stats-json [directory]',
-      'Write Webpack Stats JSON to disk',
-      resolvePathInStorybookCache(`public/`)
-    )
+    .option('--webpack-stats-json [directory]', 'Write Webpack Stats JSON to disk')
```
Comment (ghengeveld) [inline lib/core-server/src/cli/dev.ts:45]: You forgot the `<directory>`

File: lib/core-server/src/cli/prod.ts [hunk 4]
```diff
@@ -37,11 +36,7 @@ export function getProdCli(packageJson: {
     .option('--docs-dll', 'Use Docs dll reference (legacy)')
     .option('--ui-dll', 'Use UI dll reference (legacy)')
     .option('--debug-webpack', 'Display final webpack configurations for debugging purposes')
-    .option(
-      '--webpack-stats-json [directory]',
-      'Write Webpack Stats JSON to disk',
-      resolvePathInStorybookCache(`public/`)
-    )
+    .option('--webpack-stats-json [directory]', 'Write Webpack Stats JSON to disk')
```
Comment (ghengeveld) [inline lib/core-server/src/cli/prod.ts:39]: You forgot `<directory>`

**Pull request merged**